monitoring/README.md

```diff
@@ -0,0 +1,178 @@
+# Setting Up Monitoring With Grafana
+
+## Overview
+
+The monitoring service on every pi is a server which, when queried, will retrieve system data from the pi. The data currently collected includes: 
+* CPU load
+* memory usage
+* disk usage
+* number of processes running
+* temperature
+
+Once retrieved, the server will respond to the query with a json object in the following format:
+```
+{
+    "time": "Sun Nov  1 08:49:14 2020"
+    "cpu_percent": 0.0, 
+    "memory_percent": 15.6, 
+    "disk_percent": 5.4, 
+    "num_pids": 81, 
+    "temp": 61.224, 
+}
+```
+
+Once each pi is set up with its own monitoring server, the monitoring client service must be set up on the VM. This client will poll all pi monitoring servers every 5 minutes by default. It will then store the data into the file monitor.log for human-readable logs, and this log file will be used for parsing and creating the data to serve to Grafana.
+
+In addition, the monitoring client service will send slack or discord alerts via webhook to the url specified in monitoring.config. Alerts will only be sent out in two cases: if an exception occurred (e.g. a pi is unreachable), or if the threshold value for any of the data categories has been exceeded.
+
+After the monitoring client service is set up, the data source service, grafana.py, must be started. This file reads in the monitor.log file from the logs directory, and creates a dictionary based on the data extracted from this file. The format of this data will correspond to Grafana's required format for timeserie data sets when the data source is SimpleJSON. For more information on this format, please consult [the SimpleJSON plugin documentation](https://github.com/grafana/simple-json-datasource). Then, a bottle web server is started on port 8081, which will be supplied to the Grafana dashboard as the link from which to collect data.
+
+The last component to monitoring is installing Grafana itself. Grafana must be installed on the VM. Once its service is started, it will run a web server on port 3000. Through this server is how the dashboard and data panels will be set up.
+
+The following sections will explain the steps to set up monitoring in detail.
+
+## Preparation
+
+The first step is to update all the pis' package repositories, install all dependencies, make all necessary directories, and copy files to the right locations.
+
+### On the VM
+
+1. Make the /usr/bin/piman directory and copy the monitoring directory there:
+
+```
+sudo mkdir /usr/bin/piman
```

Comment:
this is not a good place to put it. perhaps /usr/local/piman/bin or /usr/share/piman/bin